app/views/movies/show.html.erb

```diff
@@ -0,0 +1,21 @@
+<h1> <%= @movie.title %> </h1>
+Year: <%= @movie.creation_year %> </br>
+Genre: <%= @movie.genre %> </br>
+
+Actors:
+<ul> <% @age_sorted_actors.each do |actor| %>
+  <div id="actor-<%= actor.id %>">
+  Name: <%= actor.name %>, Age: <%= actor.age %> 
+  </div>
+<% end %> </ul>
+
+<div id="average_age">
+Average Age of Actors in this Movie: <%= @movie.actors.average_age %>
+</div>
+
+<%= form_with url: "/movie_actors", method: :post, data: {turbo: false} do |form| %>
+  <%= form.label "Add an Actor" %>
+  <%= form.text_area :actor_id %>
+  <%= form.hidden_field :movie_id, value: "#{@movie.id}" %>
+  <%= form.submit "Submit" %>
+<% end %>
```

Comment:
Please don't forget that this is an HTML page and all of the text in this file should be wrapped in some form of HTML... currently you just have text floating around unwrapped. (ie. Year:, Genre:, etc...)

In your form, while the path that you are using is technically RESTful, doing it the way you have it causes you to create a hidden field in order to pass in the movie ID, this is something we don't want to have to do. In order to do this properly and not have to make use of the hidden field, use a nested route. I'm not taking points because we haven't spoken about nested routes yet. Here is an example of what the form would look like with a nested route:

```erb
<%= form_with url: "/movies/#{@movie.id}/actors", method: :post, data: { turbo: false } do |form| %>
  <%= form.label :actor_id, "Add an Actor" %>
  <%= form.text_field :actor_id %> # <== text field, not text area

  <%= f.submit "Submit" %>
<% end %>
```

In your controller, you would get the movie ID from the route and the actor ID from the form.

Reply:
I like this way much better. Glad we are learning nested routes today so I can do this going forward.